src/system.cpp

```diff
@@ -24,10 +42,95 @@ int SYS_initSys(void)
     memset(SYS_deviceID, 0, 32);
     strncpy(SYS_deviceID, System.deviceID().c_str(), 31);
 
+    
+
+    SYS_initTasks();
     SYS_initGPS();
+    SYS_initNVRAM();
+    SYS_initLEDs();
+
+    return 1;
+}
+
+static int SYS_initTasks(void)
+{
+    pinMode(STAT_PIN, INPUT);
+    pinMode(SF_USB_PWR_DETECT_PIN, INPUT);
+    systemFlags.hasCharger = true;
+    systemFlags.batteryLow = false;
+
+    systemDesc.pChargerCheck = &chargerTimer;
+    ledTimer.start();
+
     return 1;
 }
 
+static int SYS_initLEDs(void)
+{
+    batteryLED.init();
+
+    ledTheme.setSignal(LED_SIGNAL_NETWORK_OFF, 0x000000, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_NETWORK_ON, SF_DUP_RGB_LED_COLOR, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_NETWORK_CONNECTING, SF_DUP_RGB_LED_COLOR, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_NETWORK_DHCP, SF_DUP_RGB_LED_COLOR, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_NETWORK_CONNECTED, SF_DUP_RGB_LED_COLOR, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_CLOUD_CONNECTING, SF_DUP_RGB_LED_COLOR, LED_PATTERN_SOLID);
+    ledTheme.setSignal(LED_SIGNAL_CLOUD_CONNECTED, SF_DUP_RGB_LED_COLOR, LED_PATTERN_BLINK, SF_DUP_RGB_LED_PERIOD);
+    ledTheme.setSignal(LED_SIGNAL_CLOUD_HANDSHAKE, SF_DUP_RGB_LED_COLOR, LED_PATTERN_BLINK, SF_DUP_RGB_LED_PERIOD);
+    
+    systemDesc.pBatteryLED = &batteryLED;
+    systemDesc.systemTheme = &ledTheme;
+    return 1;
+}
+
+
+void SYS_chargerTask(void)
+{
+    bool isCharging = ~digitalRead(STAT_PIN);
+    systemFlags.hasCharger = digitalRead(SF_USB_PWR_DETECT_PIN);
```

Comment:
Neither of these seem to behave properly!

Reply:
See https://docs.particle.io/reference/device-os/api/system-calls/batterystate/

Reply:
Switched over the charging check to using the battery state, and has charger should now work with the correct pin